fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoActivity.java

```diff
@@ -145,11 +145,18 @@ private void observeToolBar() {
     binding.laoAppBar.setOnMenuItemClickListener(
         menuItem -> {
           if (menuItem.getItemId() == R.id.history_menu_toolbar) {
-            setCurrentFragment(
-                getSupportFragmentManager(),
-                R.id.fragment_digital_cash_history,
-                DigitalCashHistoryFragment::newInstance);
             binding.laoNavigationDrawer.setCheckedItem(MainMenuTab.DIGITAL_CASH.getMenuId());
+            // If the user clicks on the button when the transaction history is
+            // already displayed, then consider it as a back button pressed
+            if (!historyFlag) {
+              setCurrentFragment(
+                  getSupportFragmentManager(),
+                  R.id.fragment_digital_cash_history,
+                  DigitalCashHistoryFragment::newInstance);
+            } else {
+              openDigitalCashTab();
+            }
+            historyFlag = !historyFlag;
```

Comment:
Using a boolean is a good solution for this but it was not directly obvious to me what it does. Maybe there is some way to check that the transaction screen is already displayed instead of using the boolean? Or probably changing the name of the boolean to `isHistorySelectedFlag` or something similar would help.

Reply:
Yeah, you're right this boolean it's not exactly the cleanest way to do it. I'll remove it and use the fragment manager to detect whether it's already opened.